### ISSUE TITLE
Property setter and getter types can't differ

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#8084]
This was reported by @srittau in https://github.com/python/mypy/issues/220#issuecomment-343742241:

It also seems that the getter and setter need to have the same type at the moment:

```python
class Foo:
    @property
    def bar(self) -> str: ...
    @bar.setter
    def bar(self, v: int) -> None: ...

Foo().bar = 3
```

This will cause mypy 0.550 to complain:

> foo.py:7: error: Incompatible types in assignment (expression has type "int", variable has type "str")

While this example is a bit silly, I have a real-world example involving setting arbitrary Mappings and getting a custom Mapping subtype.

**Comment by ilevkivskyi** [comment]
Closing as duplicate of https://github.com/python/mypy/issues/3004 that has more discussion and typical use cases (non-optional getters and container normalization).
